apps/src/templates/sectionProgress/progressTables/ProgressTableLevelBubble.jsx

```diff
@@ -0,0 +1,180 @@
+    margin: '3px 0px',
+    position: 'relative'
+  },
+  largeDiamond: {
+    width: progressStyles.DIAMOND_DOT_SIZE,
+    height: progressStyles.DIAMOND_DOT_SIZE,
+    borderRadius: 4,
+    transform: 'rotate(45deg)',
+    margin: '6px 0px'
+  },
+  small: {
+    ...progressStyles.inlineBlock,
+    width: progressStyles.LETTER_BUBBLE_SIZE,
+    height: progressStyles.LETTER_BUBBLE_SIZE,
+    borderRadius: progressStyles.LETTER_BUBBLE_SIZE,
+    lineHeight: '12px',
+    fontSize: 12,
+    margin: progressStyles.LETTER_BUBBLE_MARGIN,
+    paddingRight: progressStyles.LETTER_BUBBLE_PAD,
+    paddingBottom: progressStyles.LETTER_BUBBLE_PAD,
+    textAlign: 'center',
+    verticalAlign: 'middle'
+  },
+  contents: {
+    whiteSpace: 'nowrap',
+    lineHeight: '16px'
+  },
+  diamondContents: {
+    // undo the rotation from the parent
+    transform: 'rotate(-45deg)'
+  },
+  bonusDisabled: {
+    backgroundColor: color.lighter_gray,
+    color: color.white
+  },
+  unpluggedContainer: {
+    ...progressStyles.flex,
+    borderRadius: 20,
+    padding: '6px 10px',
+    margin: '3px 0px',
+    position: 'relative'
+  },
+  unpluggedText: {
+    ...progressStyles.font,
+    fontSize: 12,
+    letterSpacing: -0.12
+  }
+};
+
+class ProgressTableLevelBubble extends React.PureComponent {
+  static propTypes = {
+    levelStatus: PropTypes.string,
+    levelKind: PropTypes.string,
+    disabled: PropTypes.bool.isRequired,
+    unplugged: PropTypes.bool,
+    smallBubble: PropTypes.bool,
+    bonus: PropTypes.bool,
+    paired: PropTypes.bool,
+    concept: PropTypes.bool,
+    title: PropTypes.string,
+    url: PropTypes.string.isRequired
+  };
+
+  levelStyle() {
+    const {levelStatus, levelKind, disabled} = this.props;
+    return {
+      ...(!disabled && progressStyles.hoverStyle),
+      ...progressStyles.levelProgressStyle(levelStatus, levelKind, disabled)
+    };
+  }
+
+  mainStyle() {
+    return {
+      ...styles.main,
+      ...this.levelStyle()
+    };
+  }
+
+  bigStyle() {
+    const {disabled, bonus, concept} = this.props;
+    return {
+      ...this.mainStyle(),
+      ...(concept && styles.largeDiamond),
+      ...(disabled && bonus && styles.bonusDisabled)
+    };
+  }
+
+  renderUnplugged() {
+    return (
+      <div style={{...styles.unpluggedContainer, ...this.levelStyle()}}>
+        <div style={styles.unpluggedText}>{i18n.unpluggedActivity()}</div>
+      </div>
+    );
+  }
+
+  renderSmallBubble() {
+    const {title} = this.props;
+    return (
+      <div>
+        <div style={{...this.mainStyle(), ...styles.small}}>{title}</div>
+      </div>
+    );
+  }
+
+  content() {
+    const {levelStatus, bonus, paired, title} = this.props;
+    const locked = levelStatus === LevelStatus.locked;
+    return locked ? (
+      <FontAwesome icon="lock" />
+    ) : paired ? (
+      <FontAwesome icon="users" />
+    ) : bonus ? (
+      <FontAwesome icon="flag-checkered" />
+    ) : (
+      (title && title) || ''
```

Comment:
Is the `title && title` intentional?

Reply:
yes: `if (title) {render title} else {render ''}` -- but this seems to be the javascripty way to do that.

Reply:
Could also do `title ? title : ''`

Reply:
oh yes, that's even clearer. i need to push a commit to trigger a drone run so i will update